lib/pdc/resource/per_thread_registry.rb

```diff
@@ -38,17 +38,19 @@ module PerThreadRegistry
     def self.extended(object)
       object.instance_variable_set '@per_thread_registry_key', object.name.freeze
     end
+
     def instance
       Thread.current[@per_thread_registry_key] ||= new
     end
+
     protected
-      def method_missing(name, *args, &block) # :nodoc:
-        # Caches the method definition as a singleton method of the receiver.
-        #
-        # By letting #delegate handle it, we avoid an enclosure that'll capture args.
-        singleton_class.delegate name, to: :instance
-        send(name, *args, &block)
-      end
+
+    def method_missing(name, *args, &block) # :nodoc:
```

Comment:
Style/MethodMissing: When using `method_missing`, define `respond_to_missing?` and fall back on `super`. (https://github.com/bbatsov/ruby-style-guide#no-method-missing)